rpm/ndb/package.go

```diff
@@ -0,0 +1,279 @@
+package ndb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash/adler32"
+	"io"
+)
+
+// Pages are hard-coded to 4096 bytes for the Package database; this
+// is different from the Index database, which could have variable page
+// sizes.
+
+// PackageDB is the "pkgdb" a.k.a. "Packages.db", the raw package data.
+type PackageDB struct {
+	r      io.ReaderAt
+	lookup map[uint32]*pkgSlot
+	slot   []pkgSlot
+	pkgHeader
+}
+
+// Parse closes over the provided [io.ReaderAt] and populates the provided PackageDB.
+func (db *PackageDB) Parse(r io.ReaderAt) error {
+	const (
+		headerSz = 4 * 8
+		pageSz   = 4096
+	)
+
+	// Read and verify the header.
+	b := make([]byte, headerSz)
+	if _, err := r.ReadAt(b, 0); err != nil {
+		return fmt.Errorf("pkg: unable to read header: %w", err)
```

Comment:
This is a nit but maybe the error messages can be a little more descriptive as to where they're being raised as `index` and `pkg` are pretty generic in clair?

Reply:
will do.